packages/backend-for-frontend/src/services/catalogService.ts

```diff
@@ -61,30 +61,35 @@ import { assertRequesterIsProducer } from "./validators.js";
 
 export type CatalogService = ReturnType<typeof catalogServiceBuilder>;
 
-const enhanceCatalogEService =
-  (
-    tenantProcessClient: TenantProcessClient,
-    agreementProcessClient: AgreementProcessClient,
-    headers: Headers,
-    requesterId: TenantId
-  ): ((eservice: catalogApi.EService) => Promise<bffApi.CatalogEService>) =>
-  async (eservice: catalogApi.EService): Promise<bffApi.CatalogEService> => {
-    const producerTenant = await tenantProcessClient.tenant.getTenant({
+const enhanceCatalogEService = (
+  tenantProcessClient: TenantProcessClient,
+  agreementProcessClient: AgreementProcessClient,
+  headers: Headers,
+  requesterId: TenantId
+): ((eservice: catalogApi.EService) => Promise<bffApi.CatalogEService>) => {
+  const tenantsCachedList: Record<string, tenantApi.Tenant> = {};
```

Comment:
```suggestion
  const tenantsCachedList: Record<TenantId, tenantApi.Tenant> = {};
```

Reply:
Done! unforunately i had to add this 
const producerTenant = ```
await getCachedTenant(
      eservice.producerId as TenantId
    );
``` because on `catalogApi.EService` producerId is a string. @Carminepo2